tests/test_strong_branching.py

```diff
@@ -0,0 +1,201 @@
+            if not self.idempotent and downinf and upinf:
+                return {"result": SCIP_RESULT.CUTOFF}
+
+            # Calculate the gains for each up and down node that strong branching explored
+            if not downinf and downvalid:
+                down_bounds[i] = down
+                downgain = max([down - lpobjval, 0])
+            else:
+                downgain = 0
+            if not upinf and upvalid:
+                up_bounds[i] = up
+                upgain = max([down - lpobjval, 0])
+            else:
+                upgain = 0
+
+            # Update the pseudo-costs
+            if not self.idempotent:
+                lpsol = branch_cands[i].getLPSol()
+                if not downinf and downvalid:
+                    self.scip.updateVarPseudoCost(branch_cands[i], -self.scip.frac(lpsol), downgain, 1)
+                if not upinf and upvalid:
+                    self.scip.updateVarPseudoCost(branch_cands[i], 1 - self.scip.frac(lpsol), upgain, 1)
+
+            scores[i] = self.scip.getBranchScoreMultiple(branch_cands[i], [downgain, upgain])
+            if scores[i] > scores[best_cand_idx]:
+                best_cand_idx = i
+
+        # End strong branching
+        self.scip.endStrongBranching()
+
+        # In the case of an LP error
+        if lperror:
+            return {"result": SCIP_RESULT.DIDNOTRUN}
+
+        # Branch on the variable with the largest score
+        down_child, eq_child, up_child = self.model.branchVarVal(
+            branch_cands[best_cand_idx], branch_cands[best_cand_idx].getLPSol())
+
+        # Update the bounds of the down node and up node
+        if self.scip.allColsInLP() and not self.idempotent:
+            if down_child is not None and down_bounds[best_cand_idx] is not None:
+                self.scip.updateNodeLowerbound(down_child, down_bounds[best_cand_idx])
+            if up_child is not None and up_bounds[best_cand_idx] is not None:
+                self.scip.updateNodeLowerbound(up_child, up_bounds[best_cand_idx])
+
+        return {"result": SCIP_RESULT.BRANCHED}
+
+
+class FeatureSelectorBranchingRule(Branchrule):
+
+    def __init__(self, scip):
+        self.scip = scip
+
+    def branchexeclp(self, allowaddcons):
+
+        if self.scip.getNNodes() == 1 or self.scip.getNNodes() == 250:
+
+            rows = self.scip.getLPRowsData()
+            cols = self.scip.getLPColsData()
+
+            # This is just a dummy rule to check functionality.
+            # A user should be able to see how they can access information without affecting the solve process.
+
+            age_row = rows[0].getAge()
+            age_col = cols[0].getAge()
+            red_cost_col = self.scip.getColRedCost(cols[0])
+
+            avg_sol = cols[0].getVar().getAvgSol()
+
+        return {"result": SCIP_RESULT.DIDNOTRUN}
+
+
+def create_model():
+    scip = Model()
+    scip.setHeuristics(SCIP_PARAMSETTING.OFF)
```

Comment:
Perhaps justify the need for disabling this?

Reply:
Added a justification